src/MahApps.Metro/MahApps.Metro.Shared/Controls/Helper/TextBoxHelper.cs

```diff
@@ -424,7 +430,41 @@ public static void SetUseFloatingWatermark(DependencyObject obj, bool value)
         {
             obj.SetValue(UseFloatingWatermarkProperty, value);
         }
-        
+
+        [Category(AppName.MahApps)]
+        [AttachedPropertyBrowsableForType(typeof(TextBoxBase))]
+        [AttachedPropertyBrowsableForType(typeof(PasswordBox))]
+        [AttachedPropertyBrowsableForType(typeof(ComboBox))]
+        [AttachedPropertyBrowsableForType(typeof(NumericUpDown))]
+        [AttachedPropertyBrowsableForType(typeof(DatePicker))]
+        [AttachedPropertyBrowsableForType(typeof(TimePickerBase))]
+        public static string GetFloatingWatermarkLocation(DependencyObject obj)
+        {
+            return obj.GetValue(FloatingWatermarkLocationProperty).ToString();
+        }
+
+        public static void SetFloatingWatermarkLocation(DependencyObject obj, FloatingWatermarkLocations value)
+        {
+            obj.SetValue(FloatingWatermarkLocationProperty, value);
+        }
+
+        [Category(AppName.MahApps)]
+        [AttachedPropertyBrowsableForType(typeof(TextBoxBase))]
+        [AttachedPropertyBrowsableForType(typeof(PasswordBox))]
+        [AttachedPropertyBrowsableForType(typeof(ComboBox))]
+        [AttachedPropertyBrowsableForType(typeof(DatePicker))]
+        [AttachedPropertyBrowsableForType(typeof(TimePickerBase))]
+        [AttachedPropertyBrowsableForType(typeof(NumericUpDown))]
+        public static string GetWatermarkTrimming(DependencyObject obj)
```

Comment:
@xxMUROxx  I will rename the `enum`, and the two functions return `string` because `FloatingWatermarkLocation` started out as `typeof string` and I forgot to change it :laughing: I'll change this too - good catch!